src/sparseml/utils/helpers.py

```diff
@@ -820,3 +823,32 @@ def json_to_jsonl(json_file_path: str, overwrite: bool = True):
         for json_line in json_data:
             json.dump(json_line, jsonl_file)  # append json line
             jsonl_file.write("\n")  # newline
+
+
+##############################
+#
+# torch helper functions
+#
+##############################
+
+
+MEMORY_BOUNDED = "MEMORY_BOUNDED"
+
+
+def memory_aware_threshold(tensor: torch.Tensor, idx: int):
+    try:
+        if (
+            MEMORY_BOUNDED in os.environ
+            and os.environ[MEMORY_BOUNDED].lower() == "true"
+        ):
+            return torch.kthvalue(tensor.view(-1), idx + 1)[0]
+        else:
+            return torch.sort(tensor.view(-1))[0][idx]
+    except RuntimeError:
+        _LOGGER.warning(
```

Comment:
thoughts on setting `MEMORY_BOUNDED=True` here so we don't run into this warning in future steps?

Reply:
great idea, thanks Ben!